include/oneapi/dpl/internal/inclusive_scan_by_segment_impl.h

```diff
@@ -106,9 +122,21 @@ inclusive_scan_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIter
                              make_zip_iterator(first2, _mask.get()) + n, make_zip_iterator(result, _mask.get()),
                              internal::segmented_scan_fun<ValueType, FlagType, BinaryOperator>(binary_op),
                              oneapi::dpl::__internal::__no_op());
-
     return result + n;
 }
+
+template <typename Policy, typename InputIterator1, typename InputIterator2, typename OutputIterator,
+          typename BinaryPredicate, typename BinaryOperator>
+oneapi::dpl::__internal::__enable_if_hetero_execution_policy<typename ::std::decay<Policy>::type, OutputIterator>
+inclusive_scan_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIterator1 last1, InputIterator2 first2,
```

Comment:
It seems like it could be pattern_nclusive_scan_by_segment and then the next level of abstraction is inclusive_scan_by_segment_impl and then scan_by_segement_impl_common. The same is true for exclusive one

Reply:
I have made this change.